sqlglot/dialects/duckdb.py

```diff
@@ -886,11 +886,14 @@ def objectinsert_sql(self, expression: exp.ObjectInsert) -> str:
             return self.func("STRUCT_INSERT", this, kv_sql)
 
         def unnest_sql(self, expression: exp.Unnest) -> str:
-            unnest_sql = super().unnest_sql(expression)
+            # Transpile BQ's UNNEST of nested array which must be made recursive to explode the top-level struct fields
+            # by transforming "FROM UNNEST(...)" to DDB's "FROM (SELECT UNNEST(..., max_depth => 2))"
+            explode_array = expression.args.get("explode_array")
+            if explode_array:
+                expression.expressions.append(
+                    exp.Kwarg(this=exp.var("max_depth"), expression=exp.Literal.number(2))
+                )
 
-            # Transpile BQ's UNNEST of nested array which must be made recursive to explode the struct fields
-            # by transforming "FROM UNNEST(...)" to DDB's "FROM (SELECT UNNEST(..., recursive => TRUE))"
-            if expression.args.get("nested_array"):
-                unnest_sql = f"(SELECT {unnest_sql[:-1]}, recursive => TRUE))"
+            unnest_sql = super().unnest_sql(expression)
 
-            return unnest_sql
+            return f"(SELECT {unnest_sql})" if explode_array else unnest_sql
```

Comment:
Can we only have `UNNEST` under `FROM/JOIN` in BigQuery? Is this wrapping always safe?

Reply:
Yeah, BQ does not allow `SELECT UNNEST(...)` as does DuckDB, and all the doc examples are based on `FROM` / `CROSS JOIN`. 

However, I think the `JOIN` side might have issues with this wrapping especially if the UNNEST is aliased. Will investigate more, hopefully we can solve both , otherwise I'll focus exclusively on `FROM`

Reply:
You can also use the builder API, e.g.

```python
if explode_array:
    select = exp.Select().subquery(unnest_alias)
    select.set("expressions", [unnest_sql])  # this is already a string so we won't have recursion issues
    unnest_sql = self.sql(select)

return unnest_sql
```

Just need to pop the unnest alias prior to generating it and saving the result in `unnest_sql` if `explode_array`. Something like that.